build-support/bin/generate_travis_yml.py

```diff
@@ -0,0 +1,771 @@
+# Env vars
+# ----------------------------------------------------------------------
+
+GLOBAL_ENV_VARS = [
+  'PANTS_CONFIG_FILES="${TRAVIS_BUILD_DIR}/pants.travis-ci.ini"',
+  'LC_ALL="en_US.UTF-8"',
+  'BOOTSTRAPPED_PEX_BUCKET=ci-public.pantsbuild.org',
+  'BOOTSTRAPPED_PEX_KEY_PREFIX=${TRAVIS_BUILD_NUMBER}/${TRAVIS_BUILD_ID}/pants.pex',
+  'BOOTSTRAPPED_PEX_URL_PREFIX=s3://${BOOTSTRAPPED_PEX_BUCKET}/${BOOTSTRAPPED_PEX_KEY_PREFIX}',
+  'PYENV_PY27_VERSION=2.7.15',
+  'PYENV_PY36_VERSION=3.6.8',
+  'PYENV_PY37_VERSION=3.7.2',
+  # NB: Linux shards use Pyenv to pre-install Python. We must not override PYENV_ROOT on those
+  # shards, or their Python will no longer work.
+  'PYENV_ROOT="${PYENV_ROOT:-${HOME}/.pants_pyenv}"',
+  'PATH="${PYENV_ROOT}/shims:${PATH}"',
+  'AWS_CLI_ROOT="${HOME}/.aws_cli"',
+  # NB: We use this verbose name so that AWS does not pick up the env var $AWS_ACCESS_KEY_ID on
+  # pull request builds. We only want this value to be populated on branch builds. Users of this
+  # env var (e.g. `deploy_to_s3.py`) are expected to re-export the env var as $AWS_ACCESS_KEY_ID.
+  'AWS_ACCESS_KEY_ID__TO_BE_REEXPORTED_ON_DEPLOYS=AKIAV6A6G7RQWPRUWIXR',
+  # This stores the encrypted AWS secret access key with the env var AWS_SECRET_ACCESS_KEY.
+  # Travis converts it back into its original decrypted value when ran in CI, per
+  # https://docs.travis-ci.com/user/environment-variables#defining-encrypted-variables-in-travisyml.
+  {"secure": "hFVAQGLVkexzTd3f9NF+JoG1dE+CPICKqOcdvQYv8+YB2rwwqu0/J6MnqKUZSmec4AM4ZvyPUBIHnSw8aMJysYs+GZ6iG/8ZRRmdbmo2WBPbSZ+ThRZxx/F6AjmovUmf8Zt366ZAZXpc9NHKREkTUGl6UL7FFe9+ouVnb90asdw="},
+  'RUST_BACKTRACE="all"',
+]
+
+# ----------------------------------------------------------------------
+# Python versions
+# ----------------------------------------------------------------------
+
+class PythonVersion(Enum):
+  py36 = "py36"
+  py37 = "py37"
+
+  def __str__(self) -> str:
+    return str(self.value)
+
+  @property
+  def number(self) -> int:
+    return {self.py36: 36, self.py37: 37}[self]
+
+  @property
+  def decimal(self) -> float:
+    return {self.py36: 3.6, self.py37: 3.7}[self]
+
+  @property
+  def is_py36(self) -> bool:
+    return self == PythonVersion.py36
+
+  @property
+  def is_py37(self) -> bool:
+    return self == PythonVersion.py37
+
+  def default_stage(self, *, is_bootstrap: bool = False) -> Stage:
+    if is_bootstrap:
+      return {self.py36: Stage.bootstrap, self.py37: Stage.bootstrap_cron}[self]
+    return {self.py36: Stage.test, self.py37: Stage.test_cron}[self]
+
+
+# ----------------------------------------------------------------------
+# AWS
+# ----------------------------------------------------------------------
+
+# We use AWS S3 to avoid unnecessary work in CI. Specifically, the bootstrap
+# shards create a pants.pex, and then upload it to S3 for all of the test
+# shards to pull down.
+
+AWS_GET_PANTS_PEX = (
```

Comment:
Could this be renamed to something like `AWS_GET_PANTS_PEX_COMMAND` (similarly for other commands) to make it more clear that this is a string as opposed to a list of arguments, or a function, or a number?